src/main/java/com/infusion/relnotesgen/ReleaseNotesModelFactory.java

```diff
@@ -54,21 +54,39 @@ public ReleaseNotesModel get() {
         Map<JiraIssueSearchType, String> errors = generateErrorMessageMap();
 		ImmutableMap<String, Issue> combinedJiraIssues = generateCombinedJiraIssues(issueIds, errors);
         Map<String, Issue> combinedJiraIssuesNoSubtasks = filterOutSubtasks(combinedJiraIssues);
-        // TODO Refactor return type
         Map<String, List<Issue>> jiraIssuesByType = issueCategorizer.byType(combinedJiraIssuesNoSubtasks.values());
 		ImmutableMap<String, ImmutableSet<ReportJiraIssueModel>> issueModelsByType = getIssuesByType(jiraIssuesByType, prMap);
 		ImmutableMap<String, ImmutableSet<ReportJiraIssueModel>> validatedIssueModelsByType = validateIssuesByType(issueModelsByType);
         ImmutableSet<ReportJiraIssueModel> knownIssues = generateKnownIssues(configuration.getKnownIssues(), errors);
         ImmutableSet<ReportJiraIssueModel> knownIssuesWithFixedIssuesRemoved = removeFixedIssuesFromKnownIssues(knownIssues, combinedJiraIssuesNoSubtasks);
-        		
-		ReleaseNotesModel model = new ReleaseNotesModel(getIssueTypes(jiraIssuesByType), validatedIssueModelsByType, 
-        		getCommitsWithDefectIds(commitsWithParsedInfo, combinedJiraIssues), knownIssuesWithFixedIssuesRemoved, versionInfoProvider.getReleaseVersion(),
-                gitInfo.commitTag1, gitInfo.commitTag2, gitInfo.commits.size(), gitInfo.gitBranch, configuration, errors);
+		ImmutableSet<ReportCommitModel> commitsWithDefectIds = getCommitsWithDefectIds(commitsWithParsedInfo);
+		ImmutableSet<ReportCommitModel> commitsWithDefectIdsFiltered = commitsWithDefectIds;//filterOutJiraIssues(commitsWithDefectIds, combinedJiraIssuesNoSubtasks);
+		ReleaseNotesModel model = new ReleaseNotesModel.ReleaseNotesModelBuilder()
+		        .issueCategoryNames(getIssueTypes(jiraIssuesByType)).issuesByCategory(validatedIssueModelsByType)
+		        .commitsWithDefectIds(commitsWithDefectIdsFiltered)
+		        .knownIssues(knownIssuesWithFixedIssuesRemoved).releaseVersion(versionInfoProvider.getReleaseVersion())
+		        .commitTag1(gitInfo.commitTag1).commitTag2(gitInfo.commitTag2).commitsCount(gitInfo.commits.size())
+		        .gitBranch(gitInfo.gitBranch).configuration(configuration).errors(errors)
+                .build();
 
         return model;
     }
 
-	private ImmutableSet<ReportJiraIssueModel> removeFixedIssuesFromKnownIssues(final ImmutableSet<ReportJiraIssueModel> knownIssues, 
+	private ImmutableSet<ReportCommitModel> filterOutJiraIssues(ImmutableSet<ReportCommitModel> commitsWithDefectIds,
```

Comment:
It's just a weird diff result.  Both methods are there.
![image](https://cloud.githubusercontent.com/assets/27444850/26706575/7a1d03a6-470c-11e7-8159-4d0bc31d387e.png)